idaes/apps/grid_integration/prescient_plugin.py

```diff
@@ -0,0 +1,524 @@
+             default='102_STEAM_3')
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--bidding',
+             help="Invoke generator strategic bidding when simulate.",
+             action='store_true',
+             dest='bidding',
+             default=False)
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--deviation-weight',
+             help="Set the weight for deviation term when tracking",
+             action='store',
+             dest='deviation_weight',
+             type='float',
+             default=30)
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--ramping-weight',
+             help="Set the weight for ramping term when tracking",
+             action='store',
+             dest='ramping_weight',
+             type='float',
+             default=20)
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--cost-weight',
+             help="Set the weight for cost term when tracking",
+             action='store',
+             dest='cost_weight',
+             type='float',
+             default=1)
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--rts_gmlc_data_dir',
+             help="the relative path to rts gmlc data set",
+             action='store',
+             dest='rts_gmlc_data_dir',
+             type='str',
+             default='./RTS-GMLC/RTS_Data/SourceData/')
+prescient.plugins.add_custom_commandline_option(opt)
+
+opt = Option('--price_forecast_file',
+             help="the relative path to price forecasts",
+             action='store',
+             dest='price_forecast_file',
+             type='str',
+             default='../../prescient/plugins/price_forecasts/lmp_forecasts_concat.csv')
+prescient.plugins.add_custom_commandline_option(opt)
+
+### End add new command line options ###
+
+from strategic_bidding import DAM_thermal_bidding
+from tracking import DAM_thermal_tracking
+
+def initialize_customized_results(options, simulator):
+
+    simulator.data_manager.extensions['customized_results'] = {}
+    customized_results = simulator.data_manager.extensions['customized_results']
+
+    customized_results['Generator'] = []
+    customized_results['Date'] = []
+    customized_results['Hour'] = []
+    customized_results['State'] = []
+    customized_results['RUC Schedule'] = []
+    customized_results['SCED Schedule'] = []
+    customized_results['Power Output'] = []
+
+    return
+prescient.plugins.register_initialization_callback(initialize_customized_results)
+
+def initialize_bidding_object(options, simulator):
+
+    '''
+    This function initializes the bidding object and stores it in the Prescient
+    data manager.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+    Returns:
+        None
+    '''
+
+    # initialize the model class
+    bidder = DAM_thermal_bidding(rts_gmlc_data_dir = options.rts_gmlc_data_dir, \
```

Comment:
How should we generalize this to support multiple user-defined IES? Should the user-defined classes be an input to this function?